core/src/main/java/org/jboss/modcluster/mcmp/impl/DefaultMCMPHandler.java

```diff
@@ -800,6 +800,16 @@ void setEstablished(boolean established) {
         private synchronized Socket getConnection() throws IOException {
             if (this.socket == null || this.socket.isClosed()) {
                 this.socket = this.socketFactory.createSocket();
+
+                // If a proxy is in ERROR or DOWN and has been configured with a hostname, as opposed to an IP address,
+                // attempt to resolve the address again. Note that JVM caches DNS name lookups. See MODCLUSTER-728.
+                if ((this.getState() != State.OK) && !this.socketAddress.isUnresolved()) {
+                    String hostName = (this.socketAddress.toString().indexOf('/') > 0) ? this.socketAddress.getHostName() : null;
```

Comment:
String hostName = (this.socketAddress.getAddress().toString().indexOf('/') > 0) ? this.socketAddress.getHostName() : null;